packages/server/modules/comments/events/emitter.ts

```diff
@@ -0,0 +1,69 @@
+import { CommentRecord } from '@/modules/comments/helpers/types'
+import { MaybeAsync } from '@/modules/shared/helpers/typeHelper'
+import { modulesDebug } from '@/modules/shared/utils/logger'
+import { EventEmitter } from 'node:events'
+
```

Comment:
@gjedlicska Besides the Redis/Bull based message queue, I've also introduced the approach of having a speckle-module-scoped (e.g. "comments" module) event emitter to allow other modules to listen for and handle certain events within that same process. Don't confuse this with the inter-process event queue we discussed previously, this only works with events fired within that same process and solves a different issue - making our code more [SOLID](https://en.wikipedia.org/wiki/SOLID).

In this PR I wanted to trigger an activity (notification) when a comment is created. While I could have just added this code to the `createComment` service call, it wouldn't be very SOLID, because it would add extra responsibilities to the call and also break the open-closed principle by requiring that devs actually modify this function to extend it.

So now, instead of modifying the `createComment` function every time we need to do something when a comment is created, you can just listen for the "COMMENT CREATED" event in some other module of yours and respond accordingly (e.g. see if it has any mentions and send out an email).

Here's an example of how this emitter is listened to: `modules/comments/services/notifications.ts`

I think this may be a useful approach to explore further as that would make our modules less coupled with each other and possibly allow custom/third-party speckle modules in the future that could use these events to do custom things.